spring-integration-core/src/test/java/org/springframework/integration/dispatcher/PollingTransactionTests.java

```diff
@@ -196,7 +200,53 @@ public void propagationMandatory() throws Throwable {
 		Message<?> errorMessage = errorChannel.receive(3000);
 		assertNotNull(errorMessage);
 		Object payload = errorMessage.getPayload();
-		assertEquals(IllegalTransactionStateException.class, payload.getClass());
+		assertEquals(MessagingException.class, payload.getClass());
+		MessagingException messagingException = (MessagingException) payload;
+		assertEquals(IllegalTransactionStateException.class, messagingException.getCause().getClass());
+		assertNull(output.receive(0));
+		assertEquals(0, txManager.getCommitCount());
+		context.close();
+	}
+
+	@Test
+	public void commitFalilureTest() throws Throwable {
+		ClassPathXmlApplicationContext context = new ClassPathXmlApplicationContext(
+				"transactionFailureTests.xml", this.getClass());
+		TestTransactionManager txManager = (TestTransactionManager) context.getBean("txManagerBad");
+		PollableChannel input = (PollableChannel) context.getBean("inputTxFailure");
+		PollableChannel output = (PollableChannel) context.getBean("output");
+		PollableChannel errorChannel = (PollableChannel) context.getBean("errorChannel");
+		assertEquals(0, txManager.getCommitCount());
+		input.send(new GenericMessage<String>("commitFalilureTest"));
+		Message<?> errorMessage = errorChannel.receive(3000);
+		assertNotNull(errorMessage);
+		Object payload = errorMessage.getPayload();
+		assertEquals(MessagingException.class, payload.getClass());
+		MessagingException messagingException = (MessagingException) payload;
+		assertEquals(IllegalTransactionStateException.class, messagingException.getCause().getClass());
+		assertNotNull(messagingException.getFailedMessage());
+		assertNotNull(output.receive(0));
+		assertEquals(0, txManager.getCommitCount());
+		context.close();
+	}
+
+	@Test
+	public void handlerFalilureTest() throws Throwable {
+		ClassPathXmlApplicationContext context = new ClassPathXmlApplicationContext(
+				"transactionFailureTests.xml", this.getClass());
```

Comment:
Can't we combine two these new tests?
I'm very sensible to any build time increases, like the same app ctx start and close. 😄 

Well, I see that is a style of this test class at all.
Would be better to revise it to a managed test with a single ctx for better performance.
But different story...
OK, let's leave your code as is for that future refactoring.